language/tools/move-package/src/compilation/compiled_bytecode.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use bytecode_source_map::source_map::SourceMap;
+use move_binary_format::file_format::{CompiledModule, CompiledScript};
+use move_core_types::{identifier::Identifier, language_storage::ModuleId};
+use move_ir_types::location::Loc;
+use move_lang::compiled_unit::CompiledUnit;
+
+#[derive(Debug, Clone)]
+pub enum ScriptOrModule {
+    Script(CompiledScript),
```

Comment:
In a similar vein to what we have today with the Diem Framework -- you may want to have scripts associated with your package for various things e.g., account creation scripts, or other various utility scripts. It makes sense to keep them in the same place, and be able to generate ABIs (and then generate transaction builders from that). 

Additionally (IMO), if we don't want end users to have to think about calling the Move compiler directly we need to support them otherwise there will be no easy way to build them.

Reply:
Ah I thought we simply want to deprecate the scripts: users should just use script functions instead.

Reply:
I don't think we want to necessarily deprecate transaction scripts -- they can be a useful feature if you want to support one-off scripts (IMO), or not have to publish a module in order to support certain scripts.